### ISSUE TITLE
Replace all does not work

### ISSUE THREAD

**Issue by N0DuckingWay** [spyder-ide/spyder#20902]
### What steps will reproduce the problem?

Simply clicking the replace all button causes spyder to have an error and exit out of the file.

### Traceback
```python-traceback
Traceback (most recent call last):
  File "/Users/zdhoffman/miniconda3/lib/python3.10/sre_parse.py", line 1051, in parse_template
    this = chr(ESCAPES[this][1])
KeyError: '\\S'

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/Users/zdhoffman/miniconda3/lib/python3.10/site-packages/spyder/widgets/findreplace.py", line 697, in replace_find_all
    cursor.insertText(re_pattern.sub(replace_text, text))
  File "/Users/zdhoffman/miniconda3/lib/python3.10/re.py", line 326, in _subx
    template = _compile_repl(template, pattern)
  File "/Users/zdhoffman/miniconda3/lib/python3.10/re.py", line 317, in _compile_repl
    return sre_parse.parse_template(repl, pattern)
  File "/Users/zdhoffman/miniconda3/lib/python3.10/sre_parse.py", line 1054, in parse_template
    raise s.error('bad escape %s' % this, len(this))
re.error: bad escape \S at position 4
```

## Versions

* Spyder version: 5.4.3  (conda)
* Python version: 3.10.10 64-bit
* Qt version: 5.15.2
* PyQt5 version: 5.15.7
* Operating System: Darwin 22.1.0

### Dependencies

```
# Mandatory:
applaunchservices >=0.3.0                        :  0.3.0 (OK)
atomicwrites >=1.2.0                             :  1.4.0 (OK)
chardet >=2.0.0                                  :  4.0.0 (OK)
cloudpickle >=0.5.0                              :  2.2.1 (OK)
cookiecutter >=1.6.0                             :  1.7.3 (OK)
diff_match_patch >=20181111                      :  20200713 (OK)
intervaltree >=3.0.2                             :  3.1.0 (OK)
IPython >=7.31.1,<9.0.0,!=8.8.0,!=8.9.0,!=8.10.0 :  8.12.0 (OK)
jedi >=0.17.2,<0.19.0                            :  0.18.1 (OK)
jellyfish >=0.7                                  :  0.9.0 (OK)
jsonschema >=3.2.0                               :  4.17.3 (OK)
keyring >=17.0.0                                 :  23.13.1 (OK)
nbconvert >=4.0                                  :  6.5.4 (OK)
numpydoc >=0.6.0                                 :  1.5.0 (OK)
parso >=0.7.0,<0.9.0                             :  0.8.3 (OK)
pexpect >=4.4.0                                  :  4.8.0 (OK)
pickleshare >=0.4                                :  0.7.5 (OK)
psutil >=5.3                                     :  5.9.0 (OK)
pygments >=2.0                                   :  2.11.2 (OK)
pylint >=2.5.0,<3.0                              :  2.16.2 (OK)
pylint_venv >=2.1.1                              :  2.3.0 (OK)
pyls_spyder >=0.4.0                              :  0.4.0 (OK)
pylsp >=1.7.2,<1.8.0                             :  1.7.2 (OK)
pylsp_black >=1.2.0                              :  1.2.1 (OK)
qdarkstyle >=3.0.2,<3.2.0                        :  3.0.2 (OK)
qstylizer >=0.2.2                                :  0.2.2 (OK)
qtawesome >=1.2.1                                :  1.2.2 (OK)
qtconsole >=5.4.2,<5.5.0                         :  5.4.2 (OK)
qtpy >=2.1.0                                     :  2.2.0 (OK)
rtree >=0.9.7                                    :  1.0.1 (OK)
setuptools >=49.6.0                              :  65.6.3 (OK)
sphinx >=0.6.6                                   :  5.0.2 (OK)
spyder_kernels >=2.4.3,<2.5.0                    :  2.4.3 (OK)
textdistance >=4.2.0                             :  4.2.1 (OK)
three_merge >=0.1.1                              :  0.1.1 (OK)
watchdog >=0.10.3                                :  2.1.6 (OK)
zmq >=22.1.0                                     :  25.0.2 (OK)

# Optional:
cython >=0.21                                    :  None (NOK)
matplotlib >=3.0.0                               :  3.7.1 (OK)
numpy >=1.7                                      :  1.24.3 (OK)
pandas >=1.1.1                                   :  1.5.3 (OK)
scipy >=0.17.0                                   :  1.10.1 (OK)
sympy >=0.7.3                                    :  None (NOK)
```

**Comment by dalthviz** [comment]
Hi @N0DuckingWay thank you for the feedback! I was unable to reproduce this (although I checked on Windows) Did you had any specific code or there is some example file that you could share with us that could help us reproduce this locally? Could it be possible to screen record (create a GIF with LICEcap for example) the way the error is triggering for you to better understand how to trigger it?

Any other info in order to reproduce this is greatly appreciated! Let us know!

**Comment by N0DuckingWay** [comment]
Here is the code that I was using, but in my experience it isn’t specific to any one file or set of code. All I have to do is open up the “find and replace” dialogue at the bottom, enter my desired values, then click the replace all button.

# -*- coding: utf-8 -*-
"""
Spyder Editor

This is a temporary script file.
"""

import fiona as fi, pandas as pd

from shapely.geometry import (Point, shape)  # shape() is a function to convert geo objects through the interface



min_distance = 30  # distance (in metres) where point is considered "on" a road
points_to_check = ((-120.2254,39.1920),(-120.2414,39.1889),(-149.1046,60.9594),(-149.0986,60.9582),(-120.22933,39.16043),
                   (-120.24767,39.15232),)  # example coordinates

traits = []

def getvals(coords):
    out = []
    for c in coords:
        if type(c) == list:
            out.extend(getvals(c))
        else:
            out.append(c)
    return out

with fi.open(r'C:\Users\zdhoffman\Documents\Apps\Ski App\Data\Input\features_skiareasonly.shp\multipolygons.shp') as shp:
    for rec in shp:
        
        boundary = shape(rec['geometry'])
        if 'osm_way_id' in dict(rec['properties']).keys():
            osm_way_id = rec['properties']['osm_way_id']
        else:
            osm_way_id = 0
        natural = rec['properties']['natural']
        landuse = rec['properties']['landuse']
        
        #
        
        contains =boundary.contains([Point(x) for x in points_to_check])
        points = [points_to_check[i] for i in range(len(points_to_check)) if contains[i]]
        if True in contains:
            
            
            traits.extend([x for x in [natural,landuse] if x != None])
            
            
            
            
                
            

traits = pd.Series(traits).value_counts()
print(traits)

**Comment by dalthviz** [comment]
Thank you for the code example @N0DuckingWay ! Indeed, I tried also the find replace functionality locally with the code provided but I'm still not able to reproduce the error. Could it be possible to also share here the values and options you are using in the find replace widget? Maybe you are using the `Use regular expressions` option?

Any other info in order to better understand this is greatly appreciated. Let us know!

**Comment by dalthviz** [comment]
Closing this since seems like a duplicate of https://github.com/spyder-ide/spyder/issues/21007 A fix for the issue will be part of Spyder 5.5.0
